### PR TITLE
Invoices with mock data

### PR DESCRIPTION
**Pull request by h-khalaf**
I have implemented GET endpoints for invoices with mock data. I have also updated error responses using a global middleware that checks error types based on the thrown classes. For now, it is used only for invoices, but it can easily be added to the rest later on.

### CODE REVIEW

**Review by LasseLejon — APPROVED** [review]

File: backend-app/src/database-Interface/database-interface-invoices.js [hunk 1]
```diff
@@ -1,12 +1,72 @@
...
+      ...invoicesValidation.getUserIDValidation(userID),
+    ];
 
-    return exports
-}
+    if (validationErrors.length > 0)
+      throw new BadRequestError(validationErrors);
+  };
+
+  exports.getInvoiceByID = (invoiceID) => {
```
Comment (fiallos00) [inline backend-app/src/database-Interface/database-interface-invoices.js:20]: Seems odd that these expect try/catch when the other database interfaces use callbacks

**Review by fiallos00 — APPROVED** [review]

File: backend-app/src/database-Interface/database-interface-invoices.js (hunk 1 above)
Comment (h-khalaf) [reply backend-app/src/database-Interface/database-interface-invoices.js:20]: These functions are going to interact with AWS S3 later on, and AWS uses promises. To make it simpler and easier to handle for the next year's students, I decided its best to have a codebase for promises :)

File: backend-app/src/index.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 const awilix = require('awilix')
+const {lifetime} = require('awilix')
```
Comment (midcoreboot) [inline backend-app/src/index.js:2]: Vet inte om jag är tokig men förstår inte varför du behöver lifetime här?

**Review by midcoreboot — APPROVED** [review]

File: backend-app/src/index.js (hunk 2 above)
Comment (h-khalaf) [reply backend-app/src/index.js:2]: Glömde att ta bort det, ska fixa det nu.

**Pull request merged**